pdg/src/main.rs

```diff
@@ -24,62 +24,111 @@ mod query;
 mod util;
 
 use builder::{construct_pdg, read_event_log};
-use clap::Parser;
+use c2rust_analysis_rt::metadata::WithMetadata;
+use clap::{Parser, ValueEnum};
 use color_eyre::eyre;
-use std::path::{Path, PathBuf};
+use std::{
+    collections::HashSet,
+    fmt::{self, Display, Formatter},
+    path::{Path, PathBuf},
+};
 
-use crate::{builder::read_metadata};
+use crate::{builder::read_metadata, graph::Func};
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, ValueEnum)]
+enum ToPrint {
+    Graphs,
+    Counts,
+    Events,
+    LatestAssignments,
+    WritePermissions,
+}
+
+impl Display for ToPrint {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        f.write_str(self.to_possible_value().unwrap().get_name())
+    }
+}
 
 /// Construct and query a PDG from an instrumented program's event log.
-#[derive(Parser, Debug)]
+#[derive(Debug, Parser)]
 #[clap(author, version, about, long_about = None)]
 struct Args {
     /// Path to an event log from a run of an instrumented program.
     #[clap(long, value_parser)]
     event_log: PathBuf,
+
     /// Path to the instrumented program's metadata generated at compile/instrumentation time.
     #[clap(long, value_parser)]
     metadata: PathBuf,
+
+    /// What to print.
+    #[clap(long, value_parser)]
+    print: Vec<ToPrint>,
```

Comment:
I had initially tried to do that because that is generally the default that we want, but I'm not sure how to do it with `clap` derive (`default_value_t = vec![ToPrint::Graphs]` doesn't work because `Vec` doesn't `impl Display`).  That said, thinking about it, if `--print graphs` is the default, how would you disable it?  I couldn't think of a good way to do that, so I think just specifying `--print graphs` is okay.  The `pdg.sh` script does it by default anyways.

Reply:
Fixed in https://github.com/immunant/c2rust/pull/493/commits/a4a5e5206d704af59bc7ee80d124600556b47272.